src/common/util/src/file_util.cpp

```diff
@@ -677,3 +691,21 @@ const char* ov::util::trim_file_name(const char* const fname) {
 #endif
     return fname_trim_ptr;
 }
+
+#ifndef __EMSCRIPTEN__
+bool ov::util::is_symlink_or_hardlink(const std::string& path) {
+    fs::path p = fs::path(path);
+    if (fs::is_symlink(p) || (fs::hard_link_count(p) > 1))
```

Comment:
`hard_link_count` return number of hard links referring to path, path `p` can be regular file path.

Why no use `std::filesystem::is_regular_file` and make alias to it like:
```cpp
namespace ov {
namespace util {
using is_regular_file = std::filesystem::is_regular_file
```